updating-bazel.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: contribute
+title: Updating Bazel
+---
+
+# Updating Bazel
+
+Bazel defined a [backwards compatibility policy](https://docs.bazel.build/versions/master/backward-compatibility.html)
+(see [https://www.bazel.build/breaking-changes-guide.html](https://www.bazel.build/breaking-changes-guide.html) for how
+to roll out incompatible changes if you are the author of one).
+This page summarized best practices on how to test and migrate your project with upcoming incompatible
+changes and how to provide feedback to the incompatible change authors.
+
+
+## Recommended migration process
+
+Bazel backwards compatibility policy is designed to avoid _upgrade cliffs_:
+any project can be prepared for the next Bazel release without breaking compatibility with the current release.
+
+We recommend the following process for project migration:
+
+
+1. Assume that your project already works with a given Bazel release, say 0.26, and you want to prepare
+   for the next release, say 0.27
+2. Find all incompatible changes for which the migration can be started: they are marked with 
+   "migration-\<release\>" label on GitHub, for example "[migration-0.26](https://github.com/bazelbuild/bazel/issues?utf8=%E2%9C%93&q=label%3Amigration-0.26+)".
+3. Each of those issues has an associated `--incompatible_*` flag. For each of them, build your project 
+   with that flag enabled, and if the build is unsuccessful, fix the project according to
+   [migration recipe](https://docs.bazel.build/versions/master/backward-compatibility.html#incompatible-changes-and-migration-recipes):
+    *   Migration guidance is available in the associated GitHub issue.
+    *   Migration is always possible in such a way that the project continues to build with and without the flag.
+    *   For some of the incompatible changes migration tooling is available, for example as part of [buildifier](https://github.com/bazelbuild/buildtools/releases). Be sure to check the GitHub issue for migration instructions.
+    *   Please report any migration problems by commenting associated GitHub issue.
+4. After all changes are migrated, you can continue to build your project without any flags:
```

Comment:
If you have a large code base and want to do a gradual migration, add the incompatible flag to your blazerc once the code is compatible.

Reply:
We should not recommend putting `--incompatible..` flags to `.bazelrc`, because they can disappear at any time.

Reply:
We do that all the time at Google. It's pretty much impossible to migrate a whole code base all at once.

Reply:
(discussed offline, here is a summary):
We can afford _just flipping the flag_ for Google code, because **no-one depends on us**.
If the code is being dependent upon by other people then maintainers of that code need to ensure that their code works _with_ and _without_ the change. If they want guarantees, ultimately they need to continuously test that, similarly to what we do on Bazel CI, where our mainline builds run without any flags, but we also have an [incompatible change pipeline](https://buildkite.com/bazel/bazelisk-plus-incompatible-flags/builds). So "put a flag to .bazelrc" is not a guidance that applies in all cases - therefore we should keep this open and let projects make individual choices. 

On top of that, we provide no guarantees for how long "--incompatible" flags are supported, so if people put those in their .bazelrc, the Bazel update will break them.